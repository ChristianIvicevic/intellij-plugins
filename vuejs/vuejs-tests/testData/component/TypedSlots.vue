<script lang="ts">
import {defineComponent, SlotsType} from 'vue'

export default defineComponent(
  {
    slots: Object as SlotsType<{
      default: { foo: string; bar: number }
      item: { data: number }
      header: { company: string }
    }>
  }
)
</script>

<template>
  <slot name="default"></slot>
  <slot name="item"></slot>
  <slot name="header"></slot>
</template>
