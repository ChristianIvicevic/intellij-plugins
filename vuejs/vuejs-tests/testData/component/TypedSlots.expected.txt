{
 class: "VueSourceComponent",
 slots: [
  {
   class: "VueSourceSlot",
   name: "default",
   scope: "{foo: string, bar: number}",
  },
  {
   class: "VueSourceSlot",
   name: "header",
   scope: "{company: string}",
  },
  {
   class: "VueSourceSlot",
   name: "item",
   scope: "{data: number}",
  },
 ],
}
